2-mandatory/3-financial-times.js

```diff
@@ -23,59 +41,83 @@ function titleWithFewestWords(allArticleTitles) {
     (Hint: remember that you can also loop through the characters of a string if you need to)
 */
 function headlinesWithNumbers(allArticleTitles) {
-    // TODO
+  // TODO
+  let titlesWithNums = [];
+
+  for (let i = 0; i < allArticleTitles.length; i++) {
+    for (x = 0; x < allArticleTitles[i].length; x++) {
+      let currentChar = allArticleTitles[i][x];
+      if (!isNaN(parseInt(currentChar))) {
+        titlesWithNums.push(allArticleTitles[i]);
+        break;
+      }
+    }
+  }
+  return titlesWithNums;
 }
 
 /*
     The Financial Times wants to understand what the average number of characters in an article title is.
     Implement the function below to return this number - rounded to the nearest integer.
 */
 function averageNumberOfCharacters(allArticleTitles) {
-    // TODO
-}
-
+  // TODO
+  let titleCharCount = [];
 
+  for (let i = 0; i < allArticleTitles.length; i++) {
+    titleCharCount.push(allArticleTitles[i].length);
+  }
+  return Math.round(
+    titleCharCount.reduce((a, b) => a + b) / titleCharCount.length
+  );
+}
```

Comment:
Hey. Excellent! You can use reduce() call. However, could you rewrite 67 - 69 in a map() call?